scalableminds/webknossos
# Include all-zero buckets in volume annotation downloads

Fixed a bug where all-zero chunks/buckets were omitted when downloading volume annotation even in case of a fallback segmentation layer, where ther zeroed-bucket information is actually needed. Otherwise, it can’t be distinguished if the area should be black or should render the fallback layer data again.

Only “reverted” buckets (marked by a single byte with value zero) are still omitted even in case of fallback segmentation layers.

### URL of deployed dev instance (used for testing):
- https://downloadvolumezeroedbuckets.webknossos.xyz

### Steps to test:
- Create volume annotation on a dataset with an existing segmentation layer (e.g. l4_sample)
- With a large brush size erase a volume comprising multiple buckets (i.e in at least 32 sections)
- Download that volume annotation, reupload
- The transparent hole should still be there, and not look like this (old behavior)
<img src="https://github.com/scalableminds/webknossos/assets/1390035/26d0d409-eb6d-4d75-8406-2825e70ef60f" width=200>

### Issues:
- fixes #7532

------
